File: javascript/space-age/space-age.js
var SpaceAge = function(seconds) {
  this.seconds = seconds;
  this.earthYearInDays = 365.25;
  this.yearLengthRelativeToEarth = {
    'earth': 1.00,
    'mercury': 0.2408467,
    'venus': 0.61519726,
    'mars': 1.8808158,
    'jupiter': 11.862615,
    'saturn': 29.447498,
    'uranus': 84.016846,
    'neptune': 164.79132
  };
};

SpaceAge.prototype.onEarth = function() {
  return this.calculateAge('earth');
};

SpaceAge.prototype.onMercury = function() {
  return this.calculateAge('mercury');
};

SpaceAge.prototype.onVenus = function() {
  return this.calculateAge('venus');
};

SpaceAge.prototype.onMars = function() {
  return this.calculateAge('mars');
};

SpaceAge.prototype.onJupiter = function() {
  return this.calculateAge('jupiter');
};

SpaceAge.prototype.onSaturn = function() {
  return this.calculateAge('saturn');
};

SpaceAge.prototype.onUranus = function() {
  return this.calculateAge('uranus');
};

SpaceAge.prototype.onNeptune = function() {
  return this.calculateAge('neptune');
};

SpaceAge.prototype.calculateAge = function(planet) {
  var planetYearInDays = this.earthYearInDays * this.yearLengthRelativeToEarth[planet];
  var planetYearInSeconds = this.convertDaysToSeconds(planetYearInDays);
  return this.ageInSeconds(planetYearInSeconds);
};

SpaceAge.prototype.convertDaysToSeconds = function(days) {
  var earthDayInSeconds = 86400;
  return days * earthDayInSeconds;
};

SpaceAge.prototype.ageInSeconds = function(planetYearInSeconds) {
  return +(this.seconds / planetYearInSeconds).toFixed(2);
};

module.exports = SpaceAge;

File: java/space-age/src/main/java/SpaceAge.java
import java.util.Map;
import java.util.HashMap;

public class SpaceAge {

    private static final long EARTH_YEAR_IN_SECONDS = 31557600;
    private static final Map<String, Double> orbitalPeriodInEarthYears = new HashMap<>();
    static {
        orbitalPeriodInEarthYears.put("Earth", 1.0);
        orbitalPeriodInEarthYears.put("Mercury", 0.2408467);
        orbitalPeriodInEarthYears.put("Venus", 0.61519726);
        orbitalPeriodInEarthYears.put("Mars", 1.8808158);
        orbitalPeriodInEarthYears.put("Jupiter", 11.862615);
        orbitalPeriodInEarthYears.put("Saturn", 29.447498);
        orbitalPeriodInEarthYears.put("Uranus", 84.016846);
        orbitalPeriodInEarthYears.put("Neptune", 164.79132);
    }
    private long seconds;

    public SpaceAge(int seconds) {
        this(Long.valueOf(seconds));
    }

    public SpaceAge(long seconds) {
        this.seconds = seconds;
    }

    public double getSeconds() {
        return (double)seconds;
    }

    public double onEarth() {
        return getSeconds() / EARTH_YEAR_IN_SECONDS;
    }

    public double onMercury() {
        return onEarth() / orbitalPeriodInEarthYears.get("Mercury");
    }

    public double onVenus() {
        return onEarth() / orbitalPeriodInEarthYears.get("Venus");
    }

    public double onMars() {
        return onEarth() / orbitalPeriodInEarthYears.get("Mars");
    }

    public double onJupiter() {
        return onEarth() / orbitalPeriodInEarthYears.get("Jupiter");
    }

    public double onSaturn() {
        return onEarth() / orbitalPeriodInEarthYears.get("Saturn");
    }

    public double onUranus() {
        return onEarth() / orbitalPeriodInEarthYears.get("Uranus");
    }

    public double onNeptune() {
        return onEarth() / orbitalPeriodInEarthYears.get("Neptune");
    }
}